test/integ/test_arrow_result.py

```diff
@@ -82,6 +96,91 @@
     "VARCHAR": [None, "ascii_test", b"\xf0\x9f\x98\x80".decode("utf-8")],
 }
 
+# Some datatypes do not match 1:1 with the original data
+PANDAS_REPRS = dict()
+if pandas_available:
+    PANDAS_REPRS = {
+        "TIMESTAMP_LTZ": [
+            pandas.NaT,
+            pandas.Timestamp("2024-01-01 12:00:00+0000", tz="UTC"),
+        ],
+        "TIMESTAMP_NTZ": [pandas.NaT, pandas.Timestamp("2024-01-01 12:00:00")],
+        "TIMESTAMP_TZ": [
+            pandas.NaT,
+            pandas.Timestamp("2024-01-01 12:00:00+0000", tz="UTC"),
+        ],
+        "NUMBER": [numpy.NAN, 1.0, 2.0, 3.0],
+    }
+
+ICEBERG_CONFIG = """
+CATALOG = 'SNOWFLAKE'
+EXTERNAL_VOLUME = 'python_connector_iceberg_exvol'
+BASE_LOCATION = 'python_connector_merge_gate';
+"""
+
+ICEBERG_UNSUPPORTED_TYPES = {
+    "CHAR",
+    "NUMBER",
+    "TIMESTAMP_TZ",
+    "INTEGER",  # Large integer example is incompatible
+}
+
+
+ICEBERG_UNSUPPORTED_NESTED_TYPES = {
+    # SNOW-1320508: Timestamp_LTZ has incorrect scale in iceberg tables
+    "TIMESTAMP_LTZ",
+}
+
+# iceberg testing is only configured in aws at the moment
+ICEBERG_ENVIRONMENTS = {"dev", "aws"}
+STRUCTRED_TYPE_ENVIRONMENTS = {"dev", "aws"}
+ICEBERG_SUPPORTED = os.getenv("cloud_provider", "dev") in ICEBERG_ENVIRONMENTS
+STRUCTURED_TYPES_SUPPORTED = (
+    os.getenv("cloud_provider", "dev") in STRUCTRED_TYPE_ENVIRONMENTS
+)
+
+PANDAS_UNSUPPORTED_NESTED_TYPES = {
+    # SNOW-1326075: Timestamp types drop information when converted to pandas
+    "TIMESTAMP_LTZ",
+    "TIMESTAMP_NTZ",
+    "TIMESTAMP_TZ",
+    "TIME",
+}
```

Comment:
shall we keep testing those types because it's something that users still encounter.
to users, it's not "unsupported" or would we block the call?